adapters/defaultLogger/defaultLogger.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2016 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package defaultLogger
+
+import (
+	"errors"
+	"os"
+	"text/template"
+
+	"istio.io/mixer"
+	"istio.io/mixer/adapters"
+)
+
+const (
+	name = "istio.io/mixer/loggers/defaultLogger"
+	desc = "Creates access log entries from Report() requests and flushes to os.Stdout"
+	tmpl = "defaultLogger"
+)
+
+type (
+	builder struct{}
+	adapter struct {
+		t *template.Template
+	}
+
+	// Config represents the adapter configuration for the default logger.
+	Config struct {
+		// Template is used to pass a golang text/template string to the
+		// adapter for translation of fact values into a log entry.
```

Comment:
I was expecting a more schematized approach by the time we hit the adapters. Basically:

- The APIHandler logic gets called with a set of facts. 

- APIHandler uses the facts to find appropriate adapter config and adapter object proper.

- For the logger, the APIHandler logic looks up the LogEntryDescriptor for the specific LogEntry, given the descriptor_name field.

- Given the LogEntryDescriptor, the APIHandler logic knows the set of labels required for the specific log entry. 

- API Handlers then fill in a new struct (AdapterLogEntry?) which has the same essential data as LogEntry, but includes the set of labels provided by the APIHandler.

- So the adapter just receives a well-formed fully-resolved AdapterLogEntry to transmit. The implementation is unaware of the greater fact model and in fact doesn't get to see the whole set of available labels.